Library/Homebrew/cmd/bottle.rb

```diff
@@ -183,6 +186,15 @@ def bottle_formula(f)
     relocatable = false
     skip_relocation = false
 
+    formula_source_time = f.stable.stage do
+      newest_mtime = nil
+      Pathname(Dir.pwd).find do |path|
+        newest_mtime ||= path.mtime
+        newest_mtime = path.mtime if path.mtime > newest_mtime
+      end
+      newest_mtime
+    end
```

Comment:
> I'd rather the performance penalty be in bottle rather than install as there's a lot more installs happening than bottles.

There isn't much penalty on install as we have to stage anyway in `install`. The only possible penalty would be scanning the file. But I highly doubt that would be a hotspot comparing to stage itself.

More importantly, this would mean network is required for bottle, which may be something we want to avoid. Especially for formulae who are huge. 

Finally, git doesn't store timestamp. So we will get different timestamp for different stage for formulae using `GitDownloadStrategy`. In such case, we should use git commit date as the timestamp. And we probably need to check other download strategies as well. Therefore, it may be more help to introduce `AbstractDownloadStrategy#newest_mtime` method. If so, putting logic in `formula_installer` make more sense.